### PR TITLE
[Edge] base code for edge event

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Update base code to handle edge event.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_query/nnstreamer_edge_common.c [hunk 1]
```diff
@@ -12,6 +12,118 @@
...
+    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+  }
+
+  ee = g_try_new (nns_edge_event_s, 1);
+  if (!ee) {
+    nns_edge_loge ("Failed to allocate memory for edge event.");
+    return NNS_EDGE_ERROR_OUT_OF_MEMORY;
+  }
+
+  memset (ee, 0, sizeof (nns_edge_event_s));
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_query/nnstreamer_edge_common.c:39]: use g_try_new0 and remove line 39? :)

File: gst/nnstreamer/tensor_query/nnstreamer_edge_common.h [hunk 2]
```diff
@@ -54,6 +54,12 @@ typedef struct {
   nns_edge_raw_data_s data[NNS_EDGE_DATA_LIMIT];
 } nns_edge_data_s;
 
+typedef struct {
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_query/nnstreamer_edge_common.h:57]: It looks like you need a lock for nns_edge_event_s. (identify and apply 'minimal' critical sections in nnstreamer_edge_common.c as well)

File: gst/nnstreamer/tensor_query/nnstreamer_edge_common.c (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_query/nnstreamer_edge_common.c:39]: I changed this, g_try_new() > malloc() as glib would be removed later. :)

File: gst/nnstreamer/tensor_query/nnstreamer_edge_common.h (hunk 2 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_query/nnstreamer_edge_common.h:68]: Marked todo item. I did not implement lock code in nns-edge yet.

**Review by gichan-jang — APPROVED** [review]
LGTM

**Pull request merged**